api/v1beta1/jujumachinetemplate_types.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type JujuMachineTemplateResource struct {
+	// Spec is the specification of the desired behavior of the machine.
+	Spec JujuMachineSpec `json:"spec"`
+}
+
+// JujuMachineTemplateSpec defines the desired state of JujuMachineTemplate
+type JujuMachineTemplateSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	Template JujuMachineTemplateResource `json:"template"`
+}
+
+// JujuMachineTemplateStatus defines the observed state of JujuMachineTemplate
+type JujuMachineTemplateStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// JujuMachineTemplate is the Schema for the jujumachinetemplates API
+type JujuMachineTemplate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   JujuMachineTemplateSpec   `json:"spec,omitempty"`
+	Status JujuMachineTemplateStatus `json:"status,omitempty"`
```

Comment:
Looking at an upstream example([link](https://cluster-api.sigs.k8s.io/developer/providers/machine-infrastructure.html#inframachinetemplate-resources)), they have no Status field for their template. Can we omit it?

Reply:
If we drop status here then I think the `//+kubebuilder:subresource:status` above will need to be removed too

Reply:
good catch, looking at some other providers (such as [talos](https://github.com/siderolabs/sidero/blob/18116bcabe77f1e7aa8d33b4d10cf2ff4a7cd18b/app/caps-controller-manager/api/v1alpha3/metalmachinetemplate_types.go#L17) and [aws](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/002bb3f57bb790a9ccd1cb054c181a6c6d532541/api/v1beta2/awsmachinetemplate_types.go#L51), they do have a status field, but that might just be because kubebuilder adds those by default. 

Since the templates arent reconciled I dont think they need a status, I will try removing things and see if anything breaks. If it goes alright I think we are safe removing the status from the template

Reply:
it looks like [CAPD](https://github.com/kubernetes-sigs/cluster-api/blob/main/test/infrastructure/docker/api/v1beta1/dockermachinetemplate_types.go) omits status, so its probably fine not to have it

Reply:
removed status in latest commit